localstack/services/kms/provider.py

```diff
@@ -404,12 +456,28 @@ def retire_grant(
     ) -> None:
         if not grant_token and (not grant_id or not key_id):
             raise ValidationException("Grant token OR (grant ID, key ID) must be specified")
-        self._delete_grant(
-            store=self._get_store(context),
-            grant_id=grant_id,
-            key_id=key_id,
-            grant_token=grant_token,
-        )
+
+        if grant_token:
+            decoded_token = to_str(base64.b64decode(grant_token))
+            grant_account_id, grant_region_name, _ = decoded_token.split(":")
+            grant_store = self._get_store(grant_account_id, grant_region_name)
+
+            if grant_token not in grant_store.grant_tokens:
+                raise NotFoundException(f"Unable to find grant token {grant_token}")
+
+            grant_id = grant_store.grant_tokens[grant_token]
+        else:
+            grant_store = self._get_store(context.account_id, context.region)
```

Comment:
This condition will never occur; in all the cases where `grant_token` is `None`, L458 will throw an exception.

Reply:
@sannya-singal: If grant token is not set, but both grant ID and key ID is set, this condition will occur.

Indeed there was no test for this, I've added one.

This can be tested with following:
```

```